core/common/src/main/java/alluxio/util/network/NettyUtils.java

```diff
@@ -168,12 +169,20 @@ public static boolean isDomainSocketSupported(WorkerNetAddress workerNetAddress)
    * @return {@link ChannelType} to use
    */
   private static ChannelType getChannelType() {
-    try {
-      EpollChannelOption.class.getField("EPOLL_MODE");
-    } catch (Throwable e) {
-      LOG.warn("EPOLL_MODE is not supported in netty with version < 4.0.26.Final.");
-      return ChannelType.NIO;
+    ChannelType configured =
+        Configuration.getEnum(PropertyKey.USER_NETWORK_NETTY_CHANNEL, ChannelType.class);
+    if (configured == ChannelType.EPOLL) {
+      if (!Epoll.isAvailable()) {
+        LOG.info("EPOLL is not available, will use NIO");
+        return ChannelType.NIO;
+      }
+      try {
+        EpollChannelOption.class.getField("EPOLL_MODE");
+      } catch (Throwable e) {
+        LOG.warn("EPOLL_MODE is not supported in netty with version < 4.0.26.Final, will use NIO");
+        return ChannelType.NIO;
+      }
     }
-    return Configuration.getEnum(PropertyKey.USER_NETWORK_NETTY_CHANNEL, ChannelType.class);
+    return configured;
```

Comment:
Should we change other places which directly query the configuration to also use `getChannelType`?

Reply:
I don't think there are any other places